mir_eval/transcription.py

```diff
@@ -138,6 +138,173 @@ def validate(ref_intervals, ref_pitches, est_intervals, est_pitches):
                          "value")
 
 
+def validate_intervals(ref_intervals, est_intervals):
+    """Checks that the input annotations to a metric look like time intervals,
+    and throws helpful errors if not.
+
+    Parameters
+    ----------
+    ref_intervals : np.ndarray, shape=(n,2)
+        Array of reference notes time intervals (onset and offset times)
+    est_intervals : np.ndarray, shape=(m,2)
+        Array of estimated notes time intervals (onset and offset times)
+    """
+    # If reference or estimated notes are empty, warn
+    if ref_intervals.size == 0:
+        warnings.warn("Reference notes are empty.")
+    if est_intervals.size == 0:
+        warnings.warn("Estimate notes are empty.")
+
+
+def match_offsets(ref_intervals, est_intervals, offset_ratio=0.2,
+                  offset_min_tolerance=0.05, strict=False):
+    """Compute a maximum matching between reference and estimated notes,
+    only taking note offsets into account.
+
```

Comment:
The function matches note offsets, hence the name. Would renaming it to `match_note_offsets` be clearer? Note offset matching (as per MIREX) has some quirky logic, which is why this is implemented as a new function as opposed to using the existing `util.match_events`